tests/integration/models/os.spec.coffee

```diff
@@ -90,14 +90,31 @@ describe 'OS model', ->
 
 		describe 'given a valid device slug', ->
 
+			expectSorted = (array, comparator) ->
+				# re-sorting could fail when the system is not using a stable
+				# sorting algorithm, in which case items of the same value
+				# might swap positions in the array
+				array.forEach (item, i) ->
+					if i == 0
+						return
+
+					previousItem = array[i - 1]
+					m.chai.expect(comparator(previousItem, item)).to.be.lte(0)
+
 			areValidVersions = (osVersions) ->
-				sortedVersions = _.clone(osVersions.versions)
-				sortedVersions.sort(osVersionRCompare)
-				return osVersions and
-					osVersions.versions and osVersions.versions.length and
-					_.isEqual(osVersions.versions, sortedVersions)
-					osVersions.latest and osVersions.recommended and osVersions.default and
-					osVersions.default is osVersions.recommended
+				m.chai.expect(osVersions).to.be.an('object')
+				m.chai.expect(osVersions).to.have.property('versions').that.is.an('array')
+				m.chai.expect(osVersions.versions).to.be.an('array')
```

Comment:
Isn't this covered on previous line? `that.is.an('array')`

Reply:
@karaxuna yea that's true :+1: I will push an update asap